File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.js
function solution(data, col, row_begin, row_end) {
    data.sort((a, b) => {
        if(a[col-1] === b[col-1]){
            return b[0] - a[0];
        }
        return a[col-1] - b[col-1];
    })
    
    var answer = 0;
    for(let i = row_begin - 1; i < row_end; i++){
        let sum = 0;
        for(const num of data[i]){
            sum += num % (i + 1);
        }
        answer ^= sum;
    }
    return answer;
}

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.java
import java.util.*;

class Solution {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        
        Arrays.sort(data, (a, b) -> {
            if(a[col-1] == b[col-1]){
                return Integer.compare(b[0], a[0]);
            }
            return Integer.compare(a[col-1], b[col-1]);
        });
        
        int answer = 0;
        for(int i = row_begin - 1; i < row_end; i++){
            int sum = 0;
            for(int num : data[i]){
                sum += num % (i + 1);
            }
            answer ^= sum;
        }
        
        return answer;
    }
}